### PR TITLE
HOSTEDCP-1556: Support Priority based expander for cluster-autoscaler

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-1556](https://issues.redhat.com/browse/HOSTEDCP-1556)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func ReconcileAutoscalerDeployment(deployment *appsv1.Deployment, hcp *hyperv1.H
 	}
 
 	args := []string{
+		"--expander=priorty,least-waste",
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go:46]: I'd expect this to be something we expose in hc.spec.ClusterAutoscaling and let it be a consumer choice.

Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go:46]: this option will be ignored if the priority configMap is not present. So the consumer choice would be to either create the configMap to use this feature or not to use the default behavior

Comment (enxebre) [reply control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go:46]: there's a typo in "priorty".

>this option will be ignored if the priority configMap is not present. So the consumer choice would be to either create the configMap to use this feature or not to use the default behavior

Yes, this enables opt-in for priority expander and fallback to least-waste. However It doesn't let a HC consumer to choose different [expanders](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-are-expanders).
I'd expect us to expose expanders in the API now to be more flexible about consumer choices e.g [cao](https://github.com/openshift/cluster-autoscaler-operator/blob/master/pkg/apis/autoscaling/v1/clusterautoscaler_types.go#L87)
That said we can always open up later and let this be our default.

Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go]: I see, makes sense.
I will create another ticket to add support for other expanders through the API.

> there's a typo in "priorty".

fixed, thanks!

**Comment by enxebre** [comment]
known issue failing kubevirt
```
    --- FAIL: TestCreateCluster/Teardown (29.55s)
        --- FAIL: TestCreateCluster/Teardown/ValidateMetricsAreExposed (0.04s)
panic: prometheus service account not found [recovered]
	panic: prometheus service account not found
```

/hold cancel 
/override ci/prow/e2e-kubevirt-aws-ovn
/cherry-pick release-4.16

**Pull request merged**